### ISSUE TITLE
Assistance needed for rewinding Git commits

### ISSUE THREAD

**Issue by Colin386** [nus-cs2113-AY2021S1/forum#62]
Is there anyway for one to rewind the commits made on the repository. I have mistakenly forgot to make parallel branches for the JavaDoc branch and the dashboard did not seem to have detected my commits for the other assignments though I have tag them correctly. Thanks

![image](https://user-images.githubusercontent.com/60383990/94004781-c72e9d80-fdcf-11ea-9b64-e167d9c929f0.png)
![image](https://user-images.githubusercontent.com/60383990/94004817-d281c900-fdcf-11ea-98b9-72a2b77eb28e.png)

**Comment by hughjazzman** [comment]
From the page:

> [This page was last updated on Sep 21 2020]

Looks like your commits were all after then. (The grading scripts have to be run, they aren’t automatic.)

**Comment by ninggggx99** [comment]
Maybe you can make some edits and create a branch to push for that?

**Comment by Colin386** [comment]
> 
> 
> From the page:
> 
> > [This page was last updated on Sep 21 2020]
> 
> Looks like your commits were all after then. (The grading scripts have to be run, they aren’t automatic.)

Oh... I see I see. No wonder the tags were not updated... Thanks for the help :).

**Comment by Colin386** [comment]
> 
> 
> Maybe you can make some edits and create a branch to push for that?

I guess I can try making a separate branch for the JavaDoc section. My concern is that it will no longer be in order with the other parts of the assignments such as the User Guide

**Comment by okkhoy** [comment]
You can:
- branch out before the JavaDocs were implemented, 
- check out the new branch
- then cherry-pick the commits related to the JavaDocs to the new branch.
- later merge it back to master. 

(The final effect would be the same since you are picking commits that are already there and merging back, but would help the script detect the parallel branch)

**Comment by Colin386** [comment]
Thanks prof.

I would like to clarify on how do we "cherry pick" the commits? Also, if I were to merge the branch will it end up merging at the very top of my master branch?

![image](https://user-images.githubusercontent.com/60383990/94006995-51c4cc00-fdd3-11ea-8706-a21410a6330c.png)

**Comment by okkhoy** [comment]
Checkout the branch-javadoc (pls follow the correct naming). Then right click on the commits you want to pick and select cherry-pick, accept the message that pops up. That is all.
Your life will be easier if you do the cherry-picking in the same order as the commits.

Next you need to push your branch to your fork and create a PR. Then you can merge the PR.
(however, I think your PR merge will say there is nothing to merge since the edits are there already -- not sure, do check)

**Comment by Colin386** [comment]
Thanks for the help Prof. And thanks for the help everyone.

I have successfully created the parallel A-JavaDoc branch and produced the tag as shown.
![image](https://user-images.githubusercontent.com/60383990/94008966-5dfe5880-fdd6-11ea-9da0-748fb88ad47b.png)

**Comment by okkhoy** [comment]
Cool!

**Comment by gmit22** [comment]
I am facing the same issue, can it somehow be done using the cli?

**Comment by okkhoy** [comment]
Yes. Please read above for the process. The commands are listed here: 

- Find out the commit id where you want to branch off
- `git checkout <commit id> -b <new branch name>`
- for each commit to be transported to the branch:
   - `git cherry-pick <commit>`
- `git push origin <new branch name>`
- `git checkout master`
- `git pull`

Create a PR and merge it.
